google/resource_compute_instance_template.go

```diff
@@ -184,23 +187,41 @@ func resourceComputeInstanceTemplate() *schema.Resource {
 				ForceNew: true,
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"network": &schema.Schema{
+						"name": &schema.Schema{
```

Comment:
Why did you add a name field here in the `network_interface` block?

Reply:
I added it because `resource_compute_instance.go` has a name field in its `network_interface` block. I was operating under the assumption the schema had to exactly match in those two resources in order for them to share a flatten/expand function.

I'm happy to remove name if that's not the case, since an instance template pretty obviously won't have an instance name.

Reply:
I managed to work around this - I only try to set the name field if the returned API object has a Name field, which instance template NICs will not.